tools/cli/tests/end-to-end.py

```diff
@@ -211,4 +216,9 @@ def test_connect(self):
 
 
 if __name__ == '__main__':
-    unittest.main()
+    for _ in range(retry_count):
```

Comment:
Does it make sense to move the retry into the test itself, so we can still rely on unittest.main() to exit with a proper error code instead of us calling sys.exit? It might require a code refactoring so a TODO is fine.

Reply:
Done.